src/python/tests/core/api/test_browser_assets.py

```diff
@@ -0,0 +1,635 @@
+        http://localhost:8082/foglamp/asset/TESTAPI/x/summary?hours=20&minutes=20&seconds=180&limit=20
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/summary?hours={}&minutes={}&seconds={}&limit={}'
+                     .format(test_data_asset_code, sensor_code_1, 20, 20, 180, 20))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 1 == len(retval)
+        assert retval[sensor_code_1]['min'] == self.test_data_x_val_list[-1]
+        assert retval[sensor_code_1]['average'] == self.test_data_x_val_list[-1]
+        assert retval[sensor_code_1]['max'] == self.test_data_x_val_list[-1]
+
+    @pytest.mark.xfail(reason="FOGL-546")
+    async def test_get_asset_sensor_readings_time_avg(self):
+        """
+        Verify that series data is grouped by default on seconds
+        http://localhost:8082/foglamp/asset/TESTAPI/x/series
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/series'.format(test_data_asset_code, sensor_code_1))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        # Find unique set of times grouped by seconds from test data
+        grouped_ts_sec = self.group_date_time(unit="second")
+
+        # Verify the length of groups and value of last element
+        assert len(grouped_ts_sec) == len(retval)
+        assert retval[-1]["average"] == self.test_data_x_val_list[-1]
+        assert retval[-1]["max"] == self.test_data_x_val_list[-1]
+        assert retval[-1]["min"] == self.test_data_x_val_list[-1]
+        assert retval[-1]["time"] == grouped_ts_sec[-1]
+
+    @pytest.mark.xfail(reason="FOGL-546")
+    async def test_get_asset_sensor_readings_time_avg_q_group_sec(self):
+        """
+        Verify that series data is grouped by seconds
+        http://localhost:8082/foglamp/asset/TESTAPI/x/series?group=seconds
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/series?group=seconds'.format(test_data_asset_code, sensor_code_1))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        # Find unique set of times grouped by seconds from test data
+        grouped_ts_sec = self.group_date_time(unit="second")
+
+        # Verify the length of groups and value of last element
+        assert len(grouped_ts_sec) == len(retval)
+        assert retval[-1]["average"] == self.test_data_x_val_list[-1]
+        assert retval[-1]["max"] == self.test_data_x_val_list[-1]
+        assert retval[-1]["min"] == self.test_data_x_val_list[-1]
+        assert retval[-1]["time"] == grouped_ts_sec[-1]
```

Comment:
only 1 record? this will not be a right test then! we should have atleast 2 records to assert  min/max/ avg calculation to be sure?

Reply:
This is the same as previous test. In this we are explicitly querying with grouping of seconds. The purpose is to verify that min/avg/max of even a single value is returned. We have further tests that asserts min/max/avg with more test data.